docs/docs/guides/gitlab.mdx

```diff
@@ -0,0 +1,435 @@
+---
+id: gitlab
+title: Gitlab Hydra integration
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl'
+
+Gitlab has several OAuth2 related features. The relevant here is the possibility
+to sign into GitLab with (almost) any OAuth2 provider, in this case hydra. The documentation for this
+is quite generic and also not very detailed. So, in this guide, we'll connect
+gitlab's omniauth-connector to ORY Hydra. We'll do that in a docker-based
+lab-environment in order to investigate the details before you do something like
+this in production.
+
+## Preparation
+
+Even though we're mostly using hydra in a docker-container, having the
+command-line-client available is quite useful. So let's install hydra as
+explained in the [installation-guide](../install).
+You'll also need [docker](https://docs.docker.com/get-docker/) and
+[docker-compose](https://docs.docker.com/compose/install/).
+
+The [hydra-5-min-tutorial](../5min-tutorial) might
+be worth checking out upfront. It'll a give a nice quick overview how OAuth2 is
+working within hydra with a minimal example. We assume basic knowledge, here.
+
+If you have not yet the source-code of hydra, which we'll need for the docker-compose
+yaml-files, clone this repository:
+
+```shell
+git clone git@github.com:ory/hydra-gitlab-example.git
```

Comment:
Is this correct? If so, we need to open source that first, right?

Reply:
No, it's not correct. Instead the user should checkout hydra. In the 5min-tutorial that is omitted and it starts with "change to the directory where the hydra-sourcecode ..." but i like the idea to include that in the Guide. Happy to also remove it completely. What do you think?

Reply:
No let's keep it here. There will be at least 10 people who do not understand that they need to run the commands (and especially which ones!) from another link. It's better to have explicit instructions.

You can move the gitlab docker-compose files to `contrib/gitlab`